samples/extensions/extended_dynamic_state2/extended_dynamic_state2.cpp

```diff
@@ -0,0 +1,1096 @@
+	    index_buffer_size,
+	    &index_staging.memory,
+	    indices.data());
+
+	cube.vertices_pos = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                        vertex_buffer_size,
+	                                                        VK_BUFFER_USAGE_VERTEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                        VMA_MEMORY_USAGE_GPU_ONLY);
+
+	cube.vertices_norm = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                         vertex_buffer_size,
+	                                                         VK_BUFFER_USAGE_VERTEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                         VMA_MEMORY_USAGE_GPU_ONLY);
+
+	cube.indices = std::make_unique<vkb::core::Buffer>(get_device(),
+	                                                   index_buffer_size,
+	                                                   VK_BUFFER_USAGE_INDEX_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
+	                                                   VMA_MEMORY_USAGE_GPU_ONLY);
+
+	/* Copy from staging buffers */
+	VkCommandBuffer copy_command = device->create_command_buffer(VK_COMMAND_BUFFER_LEVEL_PRIMARY, true);
+
+	VkBufferCopy copy_region = {};
+
+	copy_region.size = vertex_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    vertex_pos_staging.buffer,
+	    cube.vertices_pos->get_handle(),
+	    1,
+	    &copy_region);
+
+	vkCmdCopyBuffer(
+	    copy_command,
+	    vertex_norm_staging.buffer,
+	    cube.vertices_norm->get_handle(),
+	    1,
+	    &copy_region);
+
+	copy_region.size = index_buffer_size;
+	vkCmdCopyBuffer(
+	    copy_command,
+	    index_staging.buffer,
+	    cube.indices->get_handle(),
+	    1,
+	    &copy_region);
+
+	device->flush_command_buffer(copy_command, queue, true);
+
+	vkDestroyBuffer(get_device().get_handle(), vertex_pos_staging.buffer, nullptr);
+	vkFreeMemory(get_device().get_handle(), vertex_pos_staging.memory, nullptr);
+	vkDestroyBuffer(get_device().get_handle(), vertex_norm_staging.buffer, nullptr);
+	vkFreeMemory(get_device().get_handle(), vertex_norm_staging.memory, nullptr);
+	vkDestroyBuffer(get_device().get_handle(), index_staging.buffer, nullptr);
+	vkFreeMemory(get_device().get_handle(), index_staging.memory, nullptr);
+}
+
+/**
+ * @fn void ExtendedDynamicState2::cube_animation(float delta_time)
+ * @brief Changing position of one z-fighting cube (visualize negative phenomenon z-fighting)
+ */
+void ExtendedDynamicState2::cube_animation(float delta_time)
+{
+	constexpr float tick_limit = 0.05;
+	constexpr float delta      = 0.05;
+	constexpr float move_step  = 0.0005;
+	static float    time_pass  = 0;
+	time_pass += delta_time;
+	static glm::vec3 translation = scene_nodes[SCENE_BASELINE_OBJ_INDEX].at(get_node_index("Cube_1", &scene_nodes[SCENE_BASELINE_OBJ_INDEX])).node->get_transform().get_translation();
```

Comment:
You could store the reference to the transform here, and then set the translation without searching for the SceneNode again.
Like this:
```
	static auto &transform = std::find_if(scene_nodes[SCENE_BASELINE_OBJ_INDEX].begin(),
	                                      scene_nodes[SCENE_BASELINE_OBJ_INDEX].end(),
	                                      [](SceneNode const &scene_node) { return scene_node.node->get_name() == "Cube_1"; })
	                             ->node->get_transform();
	static auto  translation = transform.get_translation();

```

Reply:
Change applied. Based on this change, I removed get_node_index(), because this is the only place where this method was called.